test/runtests.jl

```diff
@@ -528,6 +528,23 @@ end
     @test sa isa StructArray
     @test axes(sa) == (-2:7,)
     @test sa.a == fill(1, -2:7)
+
+    zero_origin(T, d) = OffsetArray{T}(undef, map(n -> 0:n-1, d))
+    sa = collect_structarray(
+        [(a = 1,), (a = 2,), (a = 3,)],
+        initializer = StructArrays.StructArrayInitializer(t -> false, zero_origin),
+    )
+    @test sa isa StructArray
+    @test collect(sa.a) == 1:3
+    @test_broken sa.a isa OffsetArray
```

Comment:
Broken due to this?

https://github.com/JuliaArrays/StructArrays.jl/blob/acb24c8bc148a1a58edaa6d7a8b12d5f90b14175/src/collect.jl#L28-L29